src/backend/src/modules/puterai/AIChatService.js

```diff
@@ -390,108 +390,121 @@ class AIChatService extends BaseService {
-                    if ( event.custom ) parameters.custom = event.custom;
+                    if (event.custom) parameters.custom = event.custom;
                 }
 
-                if ( parameters.messages ) {
+                if (parameters.messages) {
                     parameters.messages =
                         Messages.normalize_messages(parameters.messages);
                 }
 
-                if ( ! test_mode && ! await this.moderate(parameters) ) {
+                if (!test_mode && ! await this.moderate(parameters)) {
                     test_mode = true;
                 }
 
-                if ( ! test_mode ) {
+                if (!test_mode) {
                     Context.set('moderated', true);
                 }
 
-                if ( test_mode ) {
+                if (test_mode) {
                     intended_service = 'fake-chat';
-                    if ( event.abuse ) {
+                    if (event.abuse) {
                         parameters.model = 'abuse';
                     }
                 }
 
-                if ( parameters.tools ) {
+                if (parameters.tools) {
                     FunctionCalling.normalize_tools_object(parameters.tools);
                 }
 
-                if ( intended_service === this.service_name ) {
-                    throw new Error('Calling ai-chat directly is not yet supported');
-                }
+                const { target_service, delegate, supplier, vendor, model } = this.dismental_model(parameters.model, intended_service);
+                // Write "target_service" back to "intended_service"
+                intended_service = target_service;
 
                 const svc_driver = this.services.get('driver');
                 let ret, error;
-                let service_used = intended_service;
-                let model_used = this.get_model_from_request(parameters, {
-                    intended_service
-                });
 
-                // Updated: Check usage and get a boolean result instead of throwing error
-                const svc_cost = this.services.get('cost');
-                const available = await svc_cost.get_available_amount();
-                
-                const model_details = this.get_model_details(model_used, {
-                    service_used,
-                });
+                const service_used = intended_service;
+
+                // Hardcode the services that accept <vendor>/<model> format. This will be
+                // removed in the future.
+                // 
+                // TODO (xiaochen): pass the whole details (supplier, vendor, model) to the
+                // delegate service, and let it decide how to handle it.
+                if (intended_service === 'openrouter') {
+                    parameters.model = vendor + '/' + model;
+                } else {
+                    parameters.model = model;
+                }
+
+                if (!delegate) {
```

Comment:
Can this be a guard clause? This is increasing the indentation of a large body of code which isn't ideal for readability and will cause additional merge conflicts